### PR TITLE
Swift 5

### PR DESCRIPTION
**Pull request by stephencelis**
Just tracking changes to Swift 5. We can keep this open for awhile.

### CODE REVIEW

File: Sources/NonEmpty/NonEmpty+Dictionary.swift [hunk 1]
```diff
@@ -106,8 +106,8 @@ extension NonEmpty where C: _DictionaryProtocol {
 
 extension NonEmpty where C: _DictionaryProtocol, C.Value: Equatable {
   public static func == (lhs: NonEmpty, rhs: NonEmpty) -> Bool {
-    return Dictionary(uniqueKeysWithValues: Array(lhs))
-      == Dictionary(uniqueKeysWithValues: Array(rhs))
+    return Dictionary<C.Key, C.Value>(uniqueKeysWithValues: Array<Element>(lhs))
```
Comment (stephencelis) [inline Sources/NonEmpty/NonEmpty+Dictionary.swift]: This now fails to compile. The type-checker can no longer infer the generics. Bug opened here: https://bugs.swift.org/browse/SR-9663

**Comment by mbrandonw** [comment]
I confirmed this builds and tests pass on Xcode 10.2 beta 👍

File: Sources/NonEmpty/NonEmpty+Dictionary.swift (hunk 1 above)
Comment (ilyapuchka) [reply Sources/NonEmpty/NonEmpty+Dictionary.swift]: Seems like swift can't properly infer types when there are labeled tuples involved. I've fixed it like this:

```swift
extension NonEmpty where C: _DictionaryProtocol, C.Value: Equatable {
  public static func == (lhs: NonEmpty, rhs: NonEmpty) -> Bool {
    let lhs = Array(lhs)
    let rhs = Array(rhs)
    return Dictionary(uniqueKeysWithValues: lhs)
      == Dictionary(uniqueKeysWithValues: rhs)
  }
}
```

https://github.com/ilyapuchka/swift-nonempty/commit/c82e104312a2539b409594faee31bc1effe67eb0#diff-b4f6b5a4e463366aef93a74e1d7a0c5c

Comment (stephencelis) [reply Sources/NonEmpty/NonEmpty+Dictionary.swift]: @ilyapuchka Are you still seeing this problem? The core team seems to have fixed the regression.

Comment (ilyapuchka) [reply Sources/NonEmpty/NonEmpty+Dictionary.swift]: I still have last year snapshot of Swift 5, so I bet this was fixed later, but still wanted to share.

**Comment by stephencelis** [comment]
Circle is failing I think because they're still on beta 3. This passes locally...

**Comment by mbrandonw** [comment]
I just pushed c4877ec to make `xcodegen` a develop dependency. look ok?

File: Sources/NonEmpty/NonEmpty+MutableCollection.swift [hunk 2]
```diff
@@ -16,6 +16,14 @@ extension NonEmpty: MutableCollection where C: MutableCollection {
         self.tail[index] = newValue
       }
     }
+    _modify {
```
Comment (mbrandonw) [inline Sources/NonEmpty/NonEmpty+MutableCollection.swift]: nice!

**Comment by stephencelis** [comment]
Merging! Will un-comment the Circle stuff when they update their images.

**Pull request merged**